feat(Authoring): Implement add lesson button menu next to lessons

geoffreykwan opened the pull request:
## Changes
- Added an add button next to all active lessons in the project view
- Clicking on the add button will open a menu that has the options "Add Lesson Before" and "Add Lesson After"
- Clicking on either of the options will create a new lesson named "New Lesson"
- The new lesson will have no steps

## Test
- Use the new add lesson button menus to add a lesson
  - before the first lesson
  - before a lesson that is not the first lesson
  - after a lesson
 - Add an unused lesson when there are no unused lessons
 - Add an active lesson when there are no active lessons

Closes #1671

geoffreykwan committed:
feat(Authoring): Implement add lesson button menu next to lessons

#1671

hirokiterashima reviewed (commented):
Functionality works. I had a few questions as I was reviewing:

1. Should we allow adding a new lesson to the unused lessons? We do this now in "Add new lesson" -> "Create your own" workflow.
2. Should the new functionality (addLessonBefore(), addLessonAfter(), and the template) be defined in a new AddNewLessonButtonComponent instead of in ProjectAuthoringComponent? It seems like the functions have nothing to do with the ProjectAuthoringComponent and therefore can be abstracted away into a separate class?

geoffreykwan committed:
feat(Authoring): Create add lesson button component

#1671

geoffreykwan committed:
feat(Authoring): Allow adding unused lesson using add lesson button menu

#1671

geoffreykwan commented:
I've made these changes
- Created add lesson button component
- Added the ability to add unused lessons using the add lesson button menu
- Added the ability to add an unused lesson when there are no unused lessons
- Added the ability to add an active lesson when there are no active lessons

geoffreykwan committed:
feat(Authoring): Use add lesson button component when no lessons

#1671

hirokiterashima commented on src/assets/wise5/authoringTool/add-lesson-button/add-lesson-button.component.html:
```diff
@@ -0,0 +1,21 @@
+<button
+  mat-icon-button
+  color="primary"
+  class="add-lesson-button"
```
Unused, remove?
```suggestion
```

hirokiterashima commented on src/assets/wise5/authoringTool/add-lesson-button/add-lesson-button.component.ts:
```diff
@@ -0,0 +1,60 @@
+  }
+
+  private createNewLesson(): any {
+    return this.projectService.createGroup('New Lesson');
```
Localize?

hirokiterashima commented on src/assets/wise5/authoringTool/project-authoring/project-authoring.component.html:
```diff
@@ -75,32 +75,51 @@
       ></project-authoring-lesson>
+      <add-lesson-button
+        class="add-lesson-button-next-to-lesson"
+        [active]="false"
```
Unnecessary, remove?
```suggestion
```

hirokiterashima commented on src/assets/wise5/authoringTool/project-authoring/project-authoring.component.html:
```diff
@@ -75,32 +75,51 @@
-    <div *ngFor="let inactiveGroupNode of inactiveGroupNodes">
+    <div *ngIf="inactiveGroupNodes.length === 0" fxLayout="row" fxLayoutAlign="start center">
+      <div i18n>There are no unused lessons</div>
+      <add-lesson-button [active]="false"></add-lesson-button>
```
Unnecessary attribute, remove?
```suggestion
      <add-lesson-button></add-lesson-button>
```

hirokiterashima reviewed (approved):
LGTM. Can you address these before merge?

- Adding a lesson before/after in the active section doesn't highlight the newly-added lesson. Highlighting new lessons seems to work in the inactive section.
- I added inline comments about code.

geoffreykwan committed:
fix(Authoring): Highlight new active lessons

#1671

geoffreykwan committed:
chore(Authoring): Remove unnecessary code

#1671
